sqs/src/main/scala/akka/stream/alpakka/sqs/SqsSourceStage.scala

```diff
@@ -61,13 +94,16 @@ final class SqsSourceStage(queueUrl: String, settings: SqsSourceSettings, sqsCli
 
       def handleSuccess(result: ReceiveMessageResult): Unit = {
 
+        currentRequests = currentRequests - 1
+        maxCurrentConcurrency = if (result.getMessages.size() == 0) 1 else maxConcurrency
+
         result.getMessages.asScala.reverse.foreach(buffer.addFirst)
```

Comment:
why isn't the `ArrayDeque` used as a FIFO `Queue`, i.e. using `offer` and `poll`?